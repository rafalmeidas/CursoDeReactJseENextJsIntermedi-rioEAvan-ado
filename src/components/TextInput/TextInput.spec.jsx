import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { TextInput } from '.';

describe('<TextInput />', () => {
  it('should have a value of inputValue', () => {
    const fn = jest.fn();
    render(<TextInput actionFn={fn} inputValue={'testando'} />);
    const input = screen.getByPlaceholderText(/type your search/i);
    expect(input).toBeInTheDocument();

    expect(input.value).toBe('testando');
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<TextInput actionFn={fn} inputValue="um valor qualquer" />);
    const input = screen.getByPlaceholderText(/type your search/i);

    const value = 'o valor';

    userEvent.type(input, value);

    expect(input.value).toBe('um valor qualquer');
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<TextInput actionFn={fn} inputValue="" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
