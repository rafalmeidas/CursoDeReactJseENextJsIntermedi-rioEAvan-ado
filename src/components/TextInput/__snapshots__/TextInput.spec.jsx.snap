// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<input
  class="text-input"
  placeholder="Type your search"
  type="text"
  value=""
/>
`;
